operator/operator_test.go

```diff
@@ -164,11 +166,25 @@ func initEnvironment(t *testing.T) *environment {
 }
 
 func (e *environment) WaitPhase() {
-	waitBlock := func() {
-		time.Sleep(blockTime * time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), blockTime*time.Duration(e.cfg.PhaseDuration+1))
```

Comment:
PhaseDuration is in number of blocks _not_ a time. This here would be a few milli seconds, so always be a done context the moment you use it.